crates/router/src/compatibility/wrap.rs

```diff
@@ -130,6 +130,20 @@ where
             .respond_to(request)
             .map_into_boxed_body()
         }
+
+        Ok(api::ApplicationResponse::PaymenkLinkForm(payment_link_data)) => {
+            match api::build_payment_link_html(*payment_link_data) {
+                Ok(rendered_html) => api::http_response_html_data(rendered_html),
+                Err(_) => api::http_response_err(
+                    r#"{
+                        "error": {
+                            "message": "Error while rendering payment link html page"
+                        }
+                    }"#,
+                ),
+            }
```

Comment:
no, since we are using a tera macro. where rendering might fail while generating the template we cannot have a string response as we have incase of build_redirection_form.